azurelinuxagent/common/protocol/util.py

```diff
@@ -204,15 +204,21 @@ def _detect_protocol(self, save_to_history, init_goal_state=True):
                     ''' 
                     # pylint: enable=W0105
                     dhcp_available = self.osutil.is_dhcp_available()
-                    if dhcp_available:
+                    # If user has DHCP disabled for their VM then the agent may enter a loop of failed dhcp requests.
```

Comment:
if conf.get_dhcp_discovery_enabled() == False, we should not reach this point

Reply:
I see, so would you recommend moving this check to get_protocol() instead? If conf.get_dhcp_discovery_enabled() == False then we never enter _detect_protocol

Reply:
The issue is _detect_protocol is what is initiating the WireProtocol instance and calling instance.detect (which created transport certs).

I could duplicate that code fragment in get_protocol